### PR TITLE
Adding the Stream Chat pod

### PR DESCRIPTION
**Pull request by MartinJibber**
Also implementing a few basic extensions for the chat client.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -9,6 +9,7 @@ def shared_pods
   pod 'TMROCoordinator'
   pod 'TMROLocalization'
 
+  pod 'StreamChatUI', '4.0.0-beta.11'
```
Comment (MartinJibber) [inline Podfile:12]: We're using the latest beta.

File: Jibber.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -16,6 +16,10 @@
 		802C448A26CC76CC00A333A4 /* Logger.swift in Sources */ = {isa = PBXBuildFile; fileRef = 802C448926CC76CC00A333A4 /* Logger.swift */; };
 		802C448B26CC76CC00A333A4 /* Logger.swift in Sources */ = {isa = PBXBuildFile; fileRef = 802C448926CC76CC00A333A4 /* Logger.swift */; };
 		802C448C26CC76CC00A333A4 /* Logger.swift in Sources */ = {isa = PBXBuildFile; fileRef = 802C448926CC76CC00A333A4 /* Logger.swift */; };
+		80616E2E26EA9E1900C077BA /* ChatClient+Extensions.swift in Sources */ = {isa = PBXBuildFile; fileRef = 80616E2D26EA9E1900C077BA /* ChatClient+Extensions.swift */; };
+		80616E2F26EA9E1900C077BA /* ChatClient+Extensions.swift in Sources */ = {isa = PBXBuildFile; fileRef = 80616E2D26EA9E1900C077BA /* ChatClient+Extensions.swift */; };
+		80616E3526EAD2F300C077BA /* DataController+Async.swift in Sources */ = {isa = PBXBuildFile; fileRef = 80616E3426EAD2F300C077BA /* DataController+Async.swift */; };
+		80616E3626EAD2F300C077BA /* DataController+Async.swift in Sources */ = {isa = PBXBuildFile; fileRef = 80616E3426EAD2F300C077BA /* DataController+Async.swift */; };
```
Comment (MartinJibber) [inline Jibber.xcodeproj/project.pbxproj:22]: Most of this PR is from the pod install. The actual code for these files is near the very bottom.

File: iOS/_Stream Temp/DataController+Async.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+//  Created by Martin Young on 9/9/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+
+extension DataController {
+
+    func synchronize() async throws {
```
Comment (MartinJibber) [inline iOS/_Stream Temp/DataController+Async.swift:14]: An Async version that all the Stream data controllers can use.

File: iOS/_Stream Temp/ChatClient+Extensions.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+//  ChatClientManager
+//
+//  Created by Martin Young on 9/9/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+
+extension ChatClient {
```
Comment (MartinJibber) [inline iOS/_Stream Temp/ChatClient+Extensions.swift:12]: No need for the chat manager any more. Just an extension with some convenience functions.

File: iOS/Flows/Home/HomeViewController.swift [hunk 5]
```diff
@@ -74,10 +74,10 @@ class HomeViewController: ViewController {
         self.collectionView.animationView.play()
 
         async let unclaimedReservationCount = Reservation.getUnclaimedReservationCount(for: User.current()!)
-        async let initialSyncFinished: Void = ConversationSupplier.shared.waitForInitialSync()
-        async let noticesLoaded: Void = NoticeSupplier.shared.loadNotices()
 
-        let _ = await (initialSyncFinished, noticesLoaded)
+        #warning("Figure out why async let isn't working here.")
+        await ConversationSupplier.shared.waitForInitialSync()
+        await NoticeSupplier.shared.loadNotices()
```
Comment (MartinJibber) [inline iOS/Flows/Home/HomeViewController.swift:80]: The async let approach I was using was crashing. Not sure why. This works for now, although it's possibly slightly slower.

File: iOS/Managers/NoticeSupplier.swift [hunk 6]
```diff
@@ -26,17 +26,19 @@ class NoticeSupplier {
 
     func loadNotices() async {
         do {
-            async let localNotices = self.getLocalNotices()
-            async let serverNotices = Notice.fetchAll()
+            #warning("Figure out why async let isn't working here")
+            let localNotices = await self.getLocalNotices()
+            let serverNotices = try await Notice.fetchAll()
```
Comment (MartinJibber) [inline iOS/Managers/NoticeSupplier.swift:31]: Same deal here. Not a huge problem at the moment though.

**Review by Benjamindodgson — APPROVED** [review]

**Pull request merged**